core/src/test/java/org/apache/iceberg/hadoop/TestHadoopCommits.java

```diff
@@ -206,6 +210,133 @@ public void testFailedCommit() throws Exception {
+    table.newFastAppend().appendFile(FILE_A).commit();
+    BaseTable baseTable = (BaseTable) table;
+    HadoopTableOperations tableOperations = (HadoopTableOperations) baseTable.operations();
+    HadoopTableOperations spyOps = spy(tableOperations);
+    doThrow(new RuntimeException("FileSystem crash!"))
+        .when(spyOps)
+        .renameMetaDataFile(any(), any(), any());
+    doThrow(new RuntimeException("Can not check new Metadata!"))
+        .when(spyOps)
+        .checkMetaDataFileRenameSuccess(any(), any(), any());
+    assertCommitNotChangeVersion(
+        baseTable, spyOps, CommitStateUnknownException.class, "FileSystem crash!");
+  }
+
+  @Test
+  public void testCommitFailedAndRenameNotSuccess() throws IOException {
+    table.newFastAppend().appendFile(FILE_A).commit();
+    BaseTable baseTable = (BaseTable) table;
+    HadoopTableOperations tableOperations = (HadoopTableOperations) baseTable.operations();
+    HadoopTableOperations spyOps = spy(tableOperations);
+    doThrow(new RuntimeException("FileSystem crash!"))
+        .when(spyOps)
+        .renameMetaDataFile(any(), any(), any());
+    doReturn(false).when(spyOps).checkMetaDataFileRenameSuccess(any(), any(), any());
+    assertCommitNotChangeVersion(
+        baseTable, spyOps, CommitFailedException.class, "Can not commit newMetaData.");
+  }
+
+  @Test
+  public void testCommitFailedButActualSuccess() throws IOException {
+    table.newFastAppend().appendFile(FILE_A).commit();
+    BaseTable baseTable = (BaseTable) table;
+    HadoopTableOperations tableOperations = (HadoopTableOperations) baseTable.operations();
+    HadoopTableOperations spyOps = spy(tableOperations);
+    doThrow(new RuntimeException("FileSystem crash!"))
+        .when(spyOps)
+        .renameMetaDataFile(any(), any(), any());
+    doReturn(true).when(spyOps).checkMetaDataFileRenameSuccess(any(), any(), any());
+    int versionBefore = spyOps.findVersion();
+    TableMetadata metadataV1 = spyOps.current();
+    SortOrder dataSort = SortOrder.builderFor(baseTable.schema()).asc("data").build();
+    TableMetadata metadataV2 = metadataV1.replaceSortOrder(dataSort);
+    spyOps.commit(metadataV1, metadataV2);
+    int versionAfter = spyOps.findVersion();
+    assert versionAfter - versionBefore == 1;
+  }
+
+  private void assertCommitNotChangeVersion(
+      BaseTable baseTable,
+      HadoopTableOperations spyOps,
+      Class<? extends RuntimeException> exceptionClass,
+      String msg) {
+    int versionBefore = spyOps.findVersion();
+    TableMetadata metadataV1 = spyOps.current();
+    SortOrder dataSort = SortOrder.builderFor(baseTable.schema()).asc("data").build();
+    TableMetadata metadataV2 = metadataV1.replaceSortOrder(dataSort);
+    assertThatThrownBy(() -> spyOps.commit(metadataV1, metadataV2))
+        .isInstanceOf(exceptionClass)
+        .hasMessageContaining(msg);
+    int versionAfter = spyOps.findVersion();
+    assert versionBefore == versionAfter;
```

Comment:
please use `assertThat()` from AssertJ to verify conditions. See also https://iceberg.apache.org/contribute/#assertj

Reply:
got it

Reply:
@nastra Sir. I modified it.